install_poetry/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: 'Install poetry'
+description: 'Installs poetry, restore python packages and optionally export requirements.txt'
+
+inputs:
+  pypi_repo_url:
+    description: 'URL of python package index (for custom packages)'
+    required: true
+
+  pypi_token_username:
+    description: 'Username for authentication at python package index (for custom packages)'
+    required: true
+
+  pypi_token:
+    description: 'Token for authentication at python package index (for custom packages)'
+    required: true
+
+  export_requirements:
+    description: "Set to 'true' if need to generate requirements.txt"
+    required: false
+    default: "false"
+
+  requirements_path:
+    description: "Destination path for requirements.txt file"
+    required: false
+    default: ".container/requirements.txt"
+
+outputs:
+  version:
+    description: "Version"
+    value: ${{ steps.release-generator.outputs.version }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install poetry and package dependencies
+      shell: bash
+      env:
+        POETRY_HTTP_BASIC_AZOPS_USERNAME: ${{ inputs.pypi_token_username }}
```

Comment:
In `POETRY_HTTP_BASIC_AZOPS_USERNAME`, `AZOPS` is the name of custom repo. I think this name should be auto-generated somehow? Or should we just use a fixed name say `CUSTOM_REPO` ?